client/src/worldgen.rs

```diff
@@ -627,4 +699,166 @@ mod test {
             epsilon = 1e-8,
         );
     }
+
+    #[test]
+    fn check_tree_planting() {
```

Comment:
I don't think this (quite large) test is actually useful, since it doesn't exercise any of the code you added/changed, and the hardcoded coordinates are clearly sound anyway. If you strip this out I'll go ahead and merge.